src/Piece.py

```diff
@@ -37,15 +47,20 @@ def movesInDirectionFromPos(self, pos, direction, side):
                         yield Move(self, newPos, pieceToCapture=pieceAtNewPos)
                     return
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Piece):
+            return NotImplemented
         if self.board == other.board and \
            self.side == other.side and \
            self.position == other.position and \
            self.__class__ == other.__class__:
             return True
         return False
 
-    def copy(self):
+    def copy(self) -> Piece:
         cpy = self.__class__(self.board, self.side, self.position,
                              movesMade=self.movesMade)
         return cpy
+
+    def getPossibleMoves(self) -> Any:
+        pass
```

Comment:
This is OK for typing. I honestly didn't noticed that this method was never defined in the parent class. Eventually it would make sense to make it an abstract method and let `Pieces` inherit from `abc.ABC`.

Reply:
> I just didn't want to introduce any major code changes as part of just adding types.

The project wasn't active for a long time, there are no tags made and the version on [PyPI](https://www.pypi.org) is really old too. I wanna change that. But before that I wanna  be sure that there are bugs fixed and that the program is a bit more stable and has more features. And if there are any more major changes, that is OK. As long as they don't break anything obviously.

Reply:
I see but if you don't mind I'd park it there with `Iterator[Move]` and add ABC to the scope of [#33]. I may be able to do some such cleanups later but can't promise it yet.